byron/chain/executable-spec/src/Cardano/Spec/Chain/STS/Rule/Chain.hs

```diff
@@ -105,6 +152,100 @@ instance Embed BBODY CHAIN where
 instance Embed PBFT CHAIN where
   wrapFailed = PBFTFailure
 
+instance Embed (UTXOWS TxId) CHAIN where
+  wrapFailed = LedgerUTXOFailure
+
+instance Embed DELEG CHAIN where
+  wrapFailed = LedgerDelegationFailure
+
 genesisHash :: Hash
 -- Not sure we need a concrete hash in the specs ...
 genesisHash = Crypto.Hash.hash ("" :: ByteString)
+
+-- | Lens for the delegation interface state contained in the chain state.
+disL :: Lens' (State CHAIN) DIState
+disL = _5
+
+instance HasTrace CHAIN where
```

Comment:
Is this trace instance composed of traces of lower level STSes?

Reply:
No, only a environment and signal generator are required. The STS rules are the ones defined compositionaly.